src/bootstrap.js

```diff
@@ -8,85 +8,102 @@ const { utils: Cu } = Components;
-    },
-    "itemDeleting": {
-      methods: ["itemDeleting"],
-      objects: ["updatingItemForm"],
-      extra_keys: ["fxauid"],
-    },
-    "itemSelected": {
-      methods: ["itemSelected"],
-      objects: ["itemList"],
-      extra_keys: ["fxauid"],
-    },
-    "addClick": {
-      methods: ["addClick"],
-      objects: ["addButton"],
-      extra_keys: ["fxauid"],
-    },
-    "itemAdded": {
-      methods: ["itemAdded"],
-      objects: ["addItemForm"],
-      extra_keys: ["itemid", "fxauid"],
-    },
-    "datastore": {
-      methods: ["added", "updated", "deleted"],
-      objects: ["datastore"],
-      extra_keys: ["itemid", "fxauid", "fields"],
-    },
-    "feedback": {
-      methods: ["feedbackClick"],
-      objects: ["manage"],
-      extra_keys: ["fxauid"],
-    },
-  });
+  try {
+    Services.telemetry.registerEvents(TELEMETRY_CATEGORY, {
+      "startup": {
+        methods: ["startup"],
+        objects: ["addon", "webextension"],
+      },
+      "iconClick": {
+        methods: ["iconClick"],
+        objects: ["toolbar"],
+        extra_keys: ["fxauid"],
+      },
+      "displayView": {
+        methods: ["render"],
+        objects: ["firstrun", "manage", "popupUnlock"],
+        extra_keys: ["fxauid"],
+      },
+      "fxaSignIn": {
+        methods: ["render"],
+        objects: ["signInPage"],
+      },
+      "confirmPW": {
+        methods: ["click"],
+        objects: ["confirmPWPage"],
+      },
+      "setupDone": {
+        methods: ["click"],
+        objects: ["setupDoneButton"],
+        extra_keys: ["fxauid"],
+      },
+      "itemAdding": {
+        methods: ["itemAdding"],
+        objects: ["addItemForm"],
+        extra_keys: ["fxauid"],
+      },
+      "itemUpdating": {
+        methods: ["itemUpdating"],
+        objects: ["updatingItemForm"],
+        extra_keys: ["fxauid"],
+      },
+      "itemDeleting": {
+        methods: ["itemDeleting"],
+        objects: ["updatingItemForm"],
+        extra_keys: ["fxauid"],
+      },
+      "itemSelected": {
+        methods: ["itemSelected"],
+        objects: ["itemList"],
+        extra_keys: ["fxauid"],
+      },
+      "addClick": {
+        methods: ["addClick"],
+        objects: ["addButton"],
+        extra_keys: ["fxauid"],
+      },
+      "itemAdded": {
+        methods: ["itemAdded"],
+        objects: ["addItemForm"],
+        extra_keys: ["itemid", "fxauid"],
+      },
+      "datastore": {
+        methods: ["added", "updated", "deleted"],
+        objects: ["datastore"],
+        extra_keys: ["itemid", "fxauid", "fields"],
+      },
+      "feedback": {
+        methods: ["feedbackClick"],
+        objects: ["manage"],
+        extra_keys: ["fxauid"],
+      },
+    });
+  } catch (e) {
+    if (e.message === "Attempt to register event that is already registered.") {
+      // eslint-disable-next-line no-console
+      console.log("telemetry events already registered; skipping registration");
+    } else {
+      throw e;
```

Comment:
I wonder if we should stop the extension from running if there are other telemetry problems.

The lack of telemetry will make @irrationalagent sad, but I don't think it's worth breaking users.  Definitely log *why*, but not sure it should stop the extension from running ...

Reply:
If there's another problem, it's (almost) certainly programmer error. The most common one would be if we used an invalid string for one of our events.